File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/IntiharIzlemFormViewModel.cs
//$86E571BE
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Infrastructure.Helpers;

namespace Core.Controllers
{
    public partial class IntiharIzlemVeriSetiServiceController
    {
        partial void PreScript_IntiharIzlemForm(IntiharIzlemFormViewModel viewModel, IntiharIzlemVeriSeti intiharIzlemVeriSeti, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionID = viewModel.GetSelectedEpisodeActionID();
            if (selectedEpisodeActionID.HasValue)
            {

                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionID.Value);


                viewModel._IntiharIzlemVeriSeti.EpisodeAction = episodeAction;
                viewModel._IntiharIzlemVeriSeti.SubEpisode = episodeAction.SubEpisode;


            }

            ContextToViewModel(viewModel, objectContext);
        }

    }
}

namespace Core.Models
{
    public partial class IntiharIzlemFormViewModel: BaseViewModel, IENabizViewModel
    {
        public void AddENabizObjectViewModelToContext(TTObjectContext objectContext)
        {
            objectContext.AddObject(this._IntiharIzlemVeriSeti);

            if (this.IntiharIzlemNedeniGridList != null) 
            {
                foreach (var item in this.IntiharIzlemNedeniGridList)
                {
                    var intiharIzlemNedeni = (IntiharIzlemNedeni)objectContext.AddObject(item);
                    intiharIzlemNedeni.IntiharIzlemVeriSeti = this._IntiharIzlemVeriSeti;
                }
            }

            if (this.IntiharIzlemYontemiGridList != null)  
            {
                foreach (var item in this.IntiharIzlemYontemiGridList)
                {
                    var intiharIzlemYontem = (IntiharIzlemYontemi)objectContext.AddObject(item);
                    intiharIzlemYontem.IntiharIzlemVeriSeti = this._IntiharIzlemVeriSeti;
                }
            }

            if (this.IntiharIzlemVakaSonucuGridList != null)
            {
                foreach (var item in this.IntiharIzlemVakaSonucuGridList)
                {
                    var intiharIzlemVakaSonucu = (IntiharIzlemVakaSonucu)objectContext.AddObject(item);
                    intiharIzlemVakaSonucu.IntiharIzlemVeriSeti = this._IntiharIzlemVeriSeti;
                }
            }





            var intiharIzlemVeriSeti = (IntiharIzlemVeriSeti)objectContext.GetLoadedObject(this._IntiharIzlemVeriSeti.ObjectID);


            new SendToENabiz(objectContext, intiharIzlemVeriSeti.SubEpisode, intiharIzlemVeriSeti.ObjectID, intiharIzlemVeriSeti.ObjectDef.Name, "229", Common.RecTime());
        }
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/IntiharIzlemFormViewModel.ts
//$07B023E0
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { IntiharIzlemVeriSeti } from "NebulaClient/Model/AtlasClientModel";
import { IntiharIzlemYontemi } from "NebulaClient/Model/AtlasClientModel";
import { IntiharIzlemVakaSonucu } from "NebulaClient/Model/AtlasClientModel";
import { IntiharIzlemNedeni } from "NebulaClient/Model/AtlasClientModel";
import { SKRSIntiharKrizVakaTuru } from "NebulaClient/Model/AtlasClientModel";
import { SKRSICD } from "NebulaClient/Model/AtlasClientModel";
import { SKRSIntiharKrizVakaSonucu } from "NebulaClient/Model/AtlasClientModel";
import { SKRSIntiharGirisimiyadaKrizNedenleri } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class IntiharIzlemFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.IntiharIzlemFormViewModel, Core';
    @Type(() => IntiharIzlemVeriSeti)
    public _IntiharIzlemVeriSeti: IntiharIzlemVeriSeti = new IntiharIzlemVeriSeti();
    @Type(() => IntiharIzlemYontemi)
    public IntiharIzlemYontemiGridList: Array<IntiharIzlemYontemi> = new Array<IntiharIzlemYontemi>();
    @Type(() => IntiharIzlemVakaSonucu)
    public IntiharIzlemVakaSonucuGridList: Array<IntiharIzlemVakaSonucu> = new Array<IntiharIzlemVakaSonucu>();
    @Type(() => IntiharIzlemNedeni)
    public IntiharIzlemNedeniGridList: Array<IntiharIzlemNedeni> = new Array<IntiharIzlemNedeni>();
    @Type(() => SKRSIntiharKrizVakaTuru)
    public SKRSIntiharKrizVakaTurus: Array<SKRSIntiharKrizVakaTuru> = new Array<SKRSIntiharKrizVakaTuru>();
    @Type(() => SKRSICD)
    public SKRSICDs: Array<SKRSICD> = new Array<SKRSICD>();
    @Type(() => SKRSIntiharKrizVakaSonucu)
    public SKRSIntiharKrizVakaSonucus: Array<SKRSIntiharKrizVakaSonucu> = new Array<SKRSIntiharKrizVakaSonucu>();
    @Type(() => SKRSIntiharGirisimiyadaKrizNedenleri)
    public SKRSIntiharGirisimiyadaKrizNedenleris: Array<SKRSIntiharGirisimiyadaKrizNedenleri> = new Array<SKRSIntiharGirisimiyadaKrizNedenleri>();
}
